### PR TITLE
fixes #27

### LINKED ISSUE #27: issue discovering #ephemeral topics
Hi again,

by using #ephemeral suffix in a channel or topicname, nsq discards message overflow (rather than write to disk) and the topic or channel disappears when there are no longer clients subscribed. 

In nsq-ruby we have issues discovering topics with the #ephemeral suffix (eg. topic name: test#ephemeral), nsq-ruby uses `URI` to connect to the nsqlookupd http interface.

looking at the code `nsq-ruby/lib/nsq/discovery.rb:73` it doesn't urlencode the `#{topic}`

With enabled debugging `nsqlookupd -verbose`  we can see that the client library just asks the `nsqlookupd` for topic "test" but it should be "test%23ephemeral"

### PR DESCRIPTION
**Pull request by mschneider82**
without urlencode

### CODE REVIEW

**Comment by mmoutenot** [comment]
wow throwback

**Comment by mschneider82** [comment]
i think there are more elegant ways to solve this issue, i am not a ruby expert.
this pull req is just a proposal which solves the problem without some 3rd party libs. 

a better solution is welcome.

File: lib/nsq/discovery.rb [hunk 1]
```diff
@@ -69,8 +69,15 @@ def get_nsqds(lookupd, topic = nil)
...
-        uri.query += "&topic=#{topic}"
+        if topic.include? '#'
+          uri.path = '/lookup'
+          topic_split = topic.split('#')
+          uri.query += "&topic=#{topic_split.first}"
+          uri.query += "%23#{topic_split.last}"
+        else
+          uri.path = '/lookup'
+          uri.query += "&topic=#{topic}"
+        end
```
Comment (alieander) [inline lib/nsq/discovery.rb]: Is nsq always going to decode the topic? if that is the case you could like solve this by simply `URI.escape`ing the topic like this:
```ruby
if topic
  uri.path = '/lookup'
  uri.query += "&topic=#{URI.escape(topic)}"
else
  uri.path = '/nodes'
end
```

**Comment by bschwartz** [comment]
@mschneider82 Thanks for this PR! Looks great. I've got a local branch going with this change as well as a test to make sure it's working and that we don't accidentally break it in the future. I'll try to get it up and merged later today or this weekend! Thanks for finding and fixing this!

**Comment by freerobby** [comment]
This looks right to me.

@bschwartz Are you thinking mainly of testing topic examples? If so, maybe add an ephemeral topic test case to discovery_spec.rb?

**Comment by bschwartz** [comment]
@freerobby Yeah, that's what I was planning to do.

**Pull request closed without merging**